source/geant4.f90

```diff
@@ -0,0 +1,145 @@
+module geant4
+
+  use floatPrecision
+  use crcoall
+  use iso_c_binding, only: C_NULL_CHAR
+  use numerical_constants
+
+  implicit none
+
+  real(kind=fPrec) :: g4_recut
+  real(kind=fPrec) :: g4_aecut
+  real(kind=fPrec) :: g4_rcut
+  real(kind=fPrec) :: g4_rangecut
+  integer :: g4_physics
+  integer :: g4_keep
+
+  character(len=64) :: phys_str
+
+  logical :: g4_enabled
+  logical :: g4_debug
+  logical :: g4_keep_stable
+
+contains
+
+subroutine geant4_fortran_init()
+
+  implicit none
+
+!Set default values
+
+!Default physics = FTFP_BERT
+  g4_physics = 0
+
+!Default ecut = none
+  g4_recut = zero
+
+!Default rcut = none
+  g4_rcut = zero
+
+!Default absecut = none
+  g4_aecut = zero
+
+  g4_rangecut = zero
+
+  g4_enabled = .false.
+
+  g4_debug = .false.
+
+  g4_keep_stable = .false.
+
+end subroutine
+
+subroutine geant4_daten(inLine,iErr)
+
+  use string_tools
+
+  implicit none
+
+  character(len=*), intent(in)    :: inLine
+  logical,          intent(inout) :: iErr
+
+  character(len=:), allocatable :: lnSplit(:)
+  integer nSplit
+  logical spErr, cErr
+
+  call chr_split(inLine,lnSplit,nSplit,spErr)
+  if(spErr) then
+    write(lout,"(a)") "GEANT4> ERROR Failed to parse input line."
+    iErr = .true.
+    return
+  end if
+
+  if(nSplit /= 2) then
```

Comment:
The line parser should return on nSplit == 0 since we allow empty lines for name/value input blocks.